src/module.c

```diff
@@ -5984,10 +5992,58 @@ RedisModuleString *RM_CreateStringFromCallReply(RedisModuleCallReply *reply) {
 }
 
 /* Modifies the user that RM_Call will use (e.g. for ACL checks) */
-void RM_SetContextUser(RedisModuleCtx *ctx, const RedisModuleUser *user) {
+void RM_SetContextUser(RedisModuleCtx *ctx, const RedisModuleUser *user)
+{
     ctx->user = user;
 }
 
+RedisModuleClient *RM_CreateModuleClient(RedisModuleCtx *ctx)
+{
+    UNUSED(ctx);
+
+    client *c = moduleAllocTempClient(NULL);
+
+    RedisModuleClient *rmc = zmalloc(sizeof(RedisModuleClient));
+    rmc->client = c;
+
+    return rmc;
+}
+
+void RM_FreeModuleClient(RedisModuleCtx *ctx, RedisModuleClient *client)
```

Comment:
Can we document the functions, specifically I think we should specify when this is OK to use this function, for example, what happened if we free the client but there is still a `RedisModuleCtx` that has this client as its `persistent_client` (can this happened?).

Reply:
so that's actually why I prevent it from being used with blocking right now, what I'd really want is reference counting on the client object to prevent that from happening (as it would happen in that case).  It be nice to be able to use it with blocking from a completeness pov, but I'm not sure of the use case (and hence why prevent it right now).